Lab_SlackBot/src/com/floreo/bbah/model/Attachment.java

```diff
@@ -12,28 +12,160 @@
 public class Attachment {
 
     // TODO: implement private fields for each of the following attachment JSON keys:
-    // "fallback"
-    // "color"
-    // "pretext"
-    // "author_name"
-    // "author_link"
-    // "author_icon"
-    // "title"
-    // "title_link"
-    // "text"
-    // "fields"
-    // "image_url"
-    // "thumb_url"
-    // "footer"
-    // "footer_icon"
-    // "ts"
+    private String fallback;
+    private String color;
+    private String pretext;
+    private String author_name;
+    private String author_link;
+    private String author_icon;
+    private String title;
+    private String title_link;
+    private String text;
+
+    // TODO: determine if we need fields
+    // Fields fields;
+
+    private String image_url;
+    private String thumb_url;
+    private String footer;
+    private String footer_icon;
+    private int ts;
 
     public Attachment(JSONObject json) {
```

Comment:
We will be using this constructor instead. I will try the parsing part once you corrected the file.